kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +274,107 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+                let parts = path.split('.').collect::<Vec<&str>>();
+                self.replace_config_with_path_parts(config_expr, &parts);
+                true
+            }
+            None => false,
         }
     }
-    pub(crate) fn _get_config_field_paths(
+
+    /// Replace AST config expr with one part of path.
+    fn replace_config_with_path_parts(
         &mut self,
-        config: &mut ast::ConfigExpr,
-    ) -> (Vec<String>, Vec<String>) {
-        let mut paths = vec![];
-        let mut paths_with_id = vec![];
-        for entry in config.items.iter_mut() {
-            let (mut _paths, mut _paths_with_id) = self._get_key_value_paths(&mut entry.node);
-            paths.append(&mut _paths);
-            paths_with_id.append(&mut &mut _paths_with_id);
+        config_expr: &mut ast::ConfigExpr,
+        parts: &[&str],
+    ) {
+        // Do not replace empty path parts and out of index parts on the config expression.
+        if parts.is_empty() {
+            return;
+        }
+        let part = parts[0];
+        let mut delete_index_set = HashSet::new();
+        // Loop all entries in the config expression and replace.
+        for (i, item) in config_expr.items.iter_mut().enumerate() {
+            // Compare each field of the config structure one by one.
+            // - `get_path_from_key` returns the real config key name.
+            // - `part` denotes the path entered by the user to be modified.
+            // For example, the real config node is `a: {b: c: {}}`. The path
+            // that needs to be modified is `a.b.c`, and its parts are ["a", "b", "c"].
+            if part == get_key_path(&item.node.key) {
+                match self.action {
+                    ast::OverrideAction::CreateOrUpdate => {
+                        // When the last part of the path is successfully recursively matched,
+                        // it indicates that the original value that needs to be overwritten
+                        // is successfully found, and the new value is used to overwrite it.
+                        // - `parts.len() == 1` denotes the path matches exactly.
+                        if parts.len() == 1 {
+                            self.override_value.set_pos(item.pos());
+                            // Override the node value.
+                            item.node.value = self.override_value.clone();
+                        } else {
+                            // Replace value recursively using the path composed by subsequent parts.
+                            //
+                            // The reason for using recursion instead of looping for path matching
+                            // is that rust cannot directly hold shared references to AST nodes
+                            // (ast::NodeRef<T> is a Box<T>), so recursive search is performed
+                            // directly on AST nodes.
+                            let config_expr = match &mut item.node.value.node {
```

Comment:
Try to extract the Schema & Config match logic to an into() func in ast crate.
Seems a little tricky, and we should reconsider the ast design later to simplify the kind of logic.

Reply:
> Try to extract the Schema & Config match logic to an into() func in ast crate. Seems a little tricky, and we should reconsider the ast design later to simplify the kind of logic.

Good suggestion. Have moved it into the `kclvm_ast` crate